src/payments-methods/express-helper/payment-methods.js

```diff
@@ -0,0 +1,128 @@
+import { settings } from '../express/settings';
+import { changeShippingAddressOptions, logOutBannerOptions } from './checkout-blocks';
+
+// Register our checkout Blocks.
+registerCheckoutBlock( changeShippingAddressOptions );
+registerCheckoutBlock( logOutBannerOptions );
+
+/**
+ * The change Payment method component.
+ *
+ * @returns React component
+ */
+const ChangePayment = () => {
+	useEffect( () => {
+		activateChange( 'amazon_change_payment_method', 'changePayment' );
+	}, [] );
+
+	return (
+		<a href="#" className="wc-apa-widget-change" id="amazon_change_payment_method">
+			{ settings.hasPaymentPreferences ? __( 'Change', 'woocommerce-gateway-amazon-payments-advanced' ) : __( 'Select', 'woocommerce-gateway-amazon-payments-advanced' ) }
+		</a>
+	);
+};
+
+/**
+ * Returns a react component and also sets an observer for the onCheckoutValidationBeforeProcessing event.
+ *
+ * @param {object} props
+ * @returns React component
+ */
+const AmazonPayInfo = ( props ) => {
+	const { shippingAddress, setShippingAddress } = props.shippingData;
+
+	const { billingData } = props.billing;
+
+	const { amazonBilling, amazonShipping } = settings.amazonAddress;
+
+	useEffect( () => {
+		const unsubscribe = props.eventRegistration.onCheckoutValidationBeforeProcessing(
+			async () => {
+				for ( const shippingField in amazonShipping ) {
+					if ( amazonShipping[ shippingField ] !== shippingAddress[ shippingField ] ) {
+						return {
+							errorMessage: __( 'We were expecting "', 'woocommerce-gateway-amazon-payments-advanced' ) + amazonShipping[ shippingField ] + __( '" but we received "', 'woocommerce-gateway-amazon-payments-advanced' ) + shippingAddress[ shippingField ] + __( '" instead for the Shipping field "', 'woocommerce-gateway-amazon-payments-advanced' ) + getCheckOutFieldsLabel( shippingField, 'shipping' ) + __( '". Please make any changes to your Shipping details through Amazon.', 'woocommerce-gateway-amazon-payments-advanced' )
+						};
+					}
+				}
+				// Not sure if we should check billing details as well. @todo
+				// if ( 'undefined' !== typeof jQuery ) {
+				// 	if ( ! jQuery( '.wc-block-checkout__use-address-for-billing input.wc-block-components-checkbox__input' ).is( ':checked' ) ) {
+				// 		for ( const billingField in amazonBilling ) {
+				// 			if ( amazonBilling[ billingField ] !== billingData[ billingField ] ) {
+				// 				return {
+				// 					errorMessage: __( 'We were expecting "', 'woocommerce-gateway-amazon-payments-advanced' ) + amazonBilling[ billingField ] + __( '" but we received "', 'woocommerce-gateway-amazon-payments-advanced' ) + billingData[ billingField ] + __( '" instead for the Billing field "', 'woocommerce-gateway-amazon-payments-advanced' ) + getCheckOutFieldsLabel( billingField, 'billing' ) + __( '". Please make any changes to your Billing details through Amazon.', 'woocommerce-gateway-amazon-payments-advanced' )
+				// 				};
+				// 			}
+				// 		}
+				// 	}
+				// }
```

Comment:
are we removing this to deploy to main repo?

Reply:
We either remove them or  we  uncomment it. I'm not 100%  sure yet,  if we  should  enforce a billing address yet. I would like to see if any issues  arise like  that from  QA. Of course, im going to note it  for it to be  tested.

Reply:
x2 to this, it's preferable to stash it imho.

Reply:
removed it, if proven we need it  can simply readd it :D